pkg/services/ngalert/models/alert_rule.go

```diff
@@ -186,6 +187,31 @@ func (alertRule *AlertRule) Diff(rule *AlertRule, ignore ...string) cmputil.Diff
 	return reporter.Diffs
 }
 
+// SetDashboardAndPanel will set the DashboardUID and PanlID
+// field be doing a lookup in the annotations. Errors when
+// the found annotations are not valid.
+func (alertRule *AlertRule) SetDashboardAndPanel() error {
+	if alertRule.Annotations == nil {
+		return nil
+	}
+	dashUID := alertRule.Annotations[DashboardUIDAnnotation]
+	panelID := alertRule.Annotations[PanelIDAnnotation]
+	if dashUID != "" && panelID == "" || dashUID == "" && panelID != "" {
```

Comment:
I would add an if statement that returns nil if `dashUid` is `""`

Reply:
Then you wouldn't catch the error if the panel is set but not the dashboard.